openthread: Add header for OpenThread module

adigie opened the pull request:
In #89090, the OpenThread implementation was moved from `net` to the `module`, but no public API header was added. As a result, it was not possible to use the implementation without including `zephyr/net/openthread.h`.

This PR adds a public API header to the OpenThread module and removes the corresponding declarations from `zephyr/net/openthread.h` in favor of including the new module header.

adigie committed:
openthread: Add header for OpenThread module

* Add header with public API of OpenThread module.
* Add module include directory to Zephyr include directories.
* Use new header in platform implementation.

Signed-off-by: Adrian Gielniewski <adrian.gielniewski@nordicsemi.no>

adigie committed:
net: openthread: Remove OpenThread module API from net header

Use OpenThread module header instead of redefining its API.

Signed-off-by: Adrian Gielniewski <adrian.gielniewski@nordicsemi.no>

rlubos reviewed (commented):
Please fix the compliance.

CC @ArekBalysNordic

adigie committed:
doc: doxygen: Add openthread module header to INPUT

Add openthread module header to INPUT due to dependencies to symbols in
`zephyr/net/openthread.h`

Signed-off-by: Adrian Gielniewski <adrian.gielniewski@nordicsemi.no>

adigie commented:
@rlubos @ArekBalysNordic I replaced usage of old API in `modules/openthread/platform/ble.c`, missed that before.
